src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -173,6 +175,16 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
   }
 
   ngOnInit() {
+    // ngx-datatable triggers calculations each time mouse enters a row,
+    // this will prevent that.
+    window.addEventListener(
+      'mouseenter',
```

Comment:
It seems `ngx-datatable` has already identified this issue https://github.com/swimlane/ngx-datatable/issues/1513, and there's a yet not merged PR to finetune this: https://github.com/swimlane/ngx-datatable/pull/1544.

If we want to fully stop parent-child propagation, we may detach [ngx-datatable from this](https://medium.com/@jtomaszewski/mastering-the-angular-performance-by-dropping-the-magic-of-change-detector-2b605b444b04).

Reply:
I saw that PR, but its open for a long time... so I tough it would be better to add that fix for now.

How would detaching fix this problem?

Reply:
We stop propagating DOM events from parent components to children? Isn't it what you're doing here for the `mouseenter` case?

Reply:
The event is triggered inside `ngx-datatable`, so even if we detach the table component, all the components inside `ngx-datatable` will still be triggered.

Now that I had another look into the code of `ngx-datatable` I will try to use the same decorator they use,  that way my listener will only encapsulate the `table.component` and not the whole window.

Reply:
Using `@HostListener` didn't work.
We have to keep the current code.